R/data-storage.R

```diff
@@ -160,12 +164,25 @@ DataStorage <- R6::R6Class( # nolint object_name_linter
             #  empty strings.
             tmp_result <- .x  %>%
               jsonlite::fromJSON() %>%
-              purrr::compact() %>%
-              as.data.frame()  %>%
+              purrr::compact()
+
+            tmp_result <- tmp_result %>%
+              purrr::map(function(.x2) {
+                if (checkmate::test_atomic_vector(.x2, min.len = 2)) {
+                  return(list(.x2))
+                }
+                if (length(.x2) > 1) {
+                  return(jsonlite::toJSON(.x2, auto_unbox = TRUE))
+                }
+                .x2
+              })
```

Comment:
I see that this anonymous function processes what we read from JSON, but it's unclear for me what is the purpose of it (mainly because I don't know the project 😅 ) but maybe if this was a named function it could describe wat we do in this step
```suggestion
              purrr::compact() %>%
              purrr::map(process_json_values_in_this_way)
```

Reply:
Moved function to `auxiliary.R`

Great feedback!